airflow/upgrade/rules/import_changes.py

```diff
@@ -49,9 +52,20 @@ def old_class(self):
     def new_class(self):
         return self.new_path.split(".")[-1]
 
+    @classmethod
+    def provider_stub_from_module(cls, module):
+        if "providers" not in module:
+            return None
+
+        # [2:] strips off the airflow.providers. part
+        parts = module.split(".")[2:]
+        if parts[0] in ('apache', 'cncf'):
```

Comment:
I think these are the only two "nested" provider folders we have, so this seemed like an easy way to get the right name for apache-cassandra or cncf-kubernetes etc.

Reply:
microsoft as well. The easiest way is to check for presence of `provider.yaml` now I think. We also have a list of providers in setup.py (with "." names) and it is verified during pre-commit.

Reply:
`setup.PROVIDERS_REQUIREMENTS.keys()`

Reply:
Ah. But this is run s part of upgrade check. I see!  It's fine - simply add "microsoft" please.